src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -314,7 +314,8 @@ private void FinishSolutionBindingOnUIThread(IProgressController controller, Can
                 return;
             }
 
-            DeterminateStepProgressNotifier progressNotifier = new DeterminateStepProgressNotifier(notificationEvents, this.BindingProjects.Count * this.NuGetPackages.Count);
+            DeterminateStepProgressNotifier progressNotifier = new DeterminateStepProgressNotifier(notificationEvents, this.BindingProjects.Count);
```

Comment:
This is working but it actually somehow hides the real problem. We were counting keys and not the proper number of nuget per project (based on the project language).